models/Asset.php

```diff
@@ -1633,6 +1633,7 @@ public function setMetadata($metadata)
         $this->setHasMetaData(false);
         if (!empty($metadata)) {
             foreach ((array)$metadata as $metaItem) {
+                $metaItem = (array)$metaItem; // also allow an object with appropriate keys (as it comes from
```

Comment:
The problem is when you have the following JSON:
```json
{
"data": {
  "metadata": [
      {
        "name": "ItemCode",
        "type": "input",
        "data": "123",
        "language": ""
      }
  ]
}
```
The metadata content in the JSON can be converted either to an associative array or to an array of stdclass objects - depending on setting `JSON_OBJECT_AS_ARRAY` as option for `json_decode` or not in https://github.com/pimcore/pimcore/blob/97a3f4de67c8c1bb35d6283b4eb06ed87f3126b3/models/Webservice/Data/Asset.php#L143-L144 and in https://github.com/pimcore/pimcore/blob/b3b1e545c0c90216ca9a89106cf3353db10ca076/lib/Tool/RestClient/AbstractRestClient.php#L408

We could add `JSON_OBJECT_AS_ARRAY` to the json_decode in https://github.com/pimcore/pimcore/blob/97a3f4de67c8c1bb35d6283b4eb06ed87f3126b3/models/Webservice/Data/Asset.php#L143 (as this only handles Asset metadata and this currently only works with an array of associative arrays).
But I do not dare to add `JSON_OBJECT_AS_ARRAY` to https://github.com/pimcore/pimcore/blob/b3b1e545c0c90216ca9a89106cf3353db10ca076/lib/Tool/RestClient/AbstractRestClient.php#L408 as this would mean that all JSON objects get converted to associative arrays - and all setters would have to to compatible with that. 

For this reason (and because Asset::setMetaData() was possible to be called with an array of stclass items before #5076 ) this PR enables allows Asset::setMetaData() to be called with an array of associative arrays or with an array of stdclass objects.